ironfish/src/network/peerNetwork.ts

```diff
@@ -896,21 +963,49 @@ export class PeerNetwork {
       return false
     }
 
+    // TODO: We may want to remove this so that transactions still propagate
+    // even with a full worker pool
     if (this.node.workerPool.saturated) {
       return false
     }
 
-    const peersToSendTo = this.connectedPeersWithoutTransaction(hash)
+    return true
+  }
 
+  alreadyHaveTransaction(hash: TransactionHash): boolean {
     /*
-     * When we receive a new transaction we want to (1) sync it to the wallet (2) add it to the mempool
-     * and (3) send it to peers who haven't received it yet. If we've already done these 3 things then
-     * return early and don't validate and verify the transaction again */
-    if (
-      this.syncedTransactionHashes.has(hash) &&
-      this.node.memPool.exists(hash) &&
-      peersToSendTo.length === 0
-    ) {
+     * When we receive a new transaction we want to test if we have already processed it yet
+     * meaning we have it in the mempool or we have it on a block. */
+
+    let peersToSendTo = 0
+    for (const _ of this.connectedPeersWithoutTransaction(hash)) {
+      peersToSendTo++
```

Comment:
If you want to make maximum use of generator, you could break out of this after hitting 1 peer, I think